src/components/logger/logger.js

```diff
@@ -0,0 +1,88 @@
+import 'typeface-montserrat';
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import image from '../../assets/settings.png';
+import { defaultFontFamily } from '../constants';
+import Content from './content';
+
+const Container = styled.div`
+  font-family: ${defaultFontFamily};
+  font-style: regular;
+  font-size: 13px;
+  background: #3f3f3f;
+  width: ${(props) => props.width}px;
+  height: ${(props) => props.height}px;
+  color: #767676;
+  position: relative;
+`;
+
+const Header = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  font-size: 13px;
+  padding: 10px;
+`;
+
+const Row = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+const Image = styled.img`
+  padding-right: 5px;
+  width: 16px;
+  height: 16px;
+`;
+
+const Title = styled.div`
+  font-size: 13px;
+  text-transform: uppercase;
+`;
+
+const Clear = styled.div`
+  cursor: pointer;
+`;
+
+export default class Logger extends React.Component {
+  static propTypes = {
+    width: PropTypes.number.isRequired,
+    height: PropTypes.number.isRequired,
+    logs: PropTypes.object.isRequired,
+  };
+
+  state = {
+    events: [],
+  };
+
+  onClear = () => this.setState({ events: [] });
+
+  componentDidMount() {
+    this.logsSubscription = this.props.logs.subscribe((event) => {
+      this.setState({ events: [...this.state.events, event] });
```

Comment:
if you rely on previous state, it's highly recommended to use the callback way of setting state

Reply:
fixed